File: internal/cli/path.go
package cli

import (
	"fmt"

	"github.com/spf13/cobra"
)

func newPathCmd(d Deps) *cobra.Command {
	return &cobra.Command{
		Use:   "path <branch> <repo>",
		Short: "Print the absolute worktree path for branch and repo",
		Long: `Resolve and print only the absolute worktree path for <repo> on <branch>.

Uses the same worktree_path template as add/rm (including §5.1 dual defaults).
Does not require the worktree to exist on disk.`,
		Args: cobra.ExactArgs(2),
		RunE: func(cmd *cobra.Command, args []string) error {
			return runPath(d, args[0], args[1])
		},
	}
}

func runPath(d Deps, branch, repo string) error {
	if branch == "" {
		return fmt.Errorf("branch is required")
	}
	if repo == "" {
		return fmt.Errorf("repo is required")
	}
	cfg, err := d.loadConfig()
	if err != nil {
		return err
	}
	// Validate repo is in config (path is still deterministic from template).
	if _, err := selectRepos(cfg, []string{repo}); err != nil {
		return err
	}
	ctx, err := resolveRepo(cfg, repo, branch)
	if err != nil {
		return err
	}
	fmt.Fprintln(d.stdout(), ctx.WorktreePath)
	return nil
}

File: src/cli/path.rs
//! `mwt path <branch> <repo>` — print the absolute worktree path.

use std::io::Write;

use clap::Args;

use crate::cli::deps::Deps;
use crate::cli::format::{OutputFormat, write_path};
use crate::cli::shared::{resolve_repo, select_repos};

#[derive(Debug, Args)]
pub struct PathOptions {
    /// Output format: `table` (default) or `json`.
    #[arg(long, value_enum, default_value_t = OutputFormat::Table)]
    pub format: OutputFormat,
}

pub fn run(
    deps: &Deps,
    branch: &str,
    repo: &str,
    opts: &PathOptions,
    out: &mut dyn Write,
) -> Result<(), String> {
    if branch.is_empty() {
        return Err("branch is required".to_string());
    }
    if repo.is_empty() {
        return Err("repo is required".to_string());
    }
    let cfg = (deps.load_config)().map_err(|e| format!("load config: {e}"))?;
    // Validate repo is in config (path is still deterministic from template).
    select_repos(&cfg, std::slice::from_ref(&repo.to_string()))?;
    let ctx = resolve_repo(&cfg, repo, branch)?;
    write_path(out, opts.format, &ctx.worktree_path.to_string_lossy())
        .map_err(|e| e.to_string())?;
    Ok(())
}
